src/Language/PureScript/Sugar/TypeClasses.hs

```diff
@@ -205,7 +205,8 @@ desugarDecl
 desugarDecl mn exps = go
   where
   go d@(TypeClassDeclaration sa name args implies deps members) = do
-    modify (M.insert (mn, name) (makeTypeClassData args (map memberToNameAndType members) implies deps False))
+    let addNullVtaArgs (a, b) = (a, b, Nothing)
+    modify (M.insert (mn, name) (makeTypeClassData args (map (addNullVtaArgs . memberToNameAndType) members) implies deps False))
```

Comment:
This part may be problematic as I have to put something here, but I don't have anything else to provide. That being said, I haven't looked into this code that much.

Reply:
AFAICT, this should not be adding `Nothing` here and instead be adding the VTA information. Latest commits push that.

Reply:
As this comment (https://github.com/purescript/purescript/pull/4513#issuecomment-1782826341) shows, putting data here causes other compile errors to arise. See the rest of the comment for a more detailed explanation.